### PR TITLE
[ET-797] RSVP FE > Admin sees update notice

### PR DESCRIPTION
**Pull request by Luc45**
Ticket: https://moderntribe.atlassian.net/browse/ET-797

_I'm setting ET-796 as the base as I'm building on top of some groundwork code that I've done there._

### CODE REVIEW

File: src/Tribe/RSVP/Early_Access/Update_Notice.php [hunk 1]
```diff
@@ -0,0 +1,87 @@
+<?php
+
```
Comment (tr1b0t) [inline src/Tribe/RSVP/Early_Access/Update_Notice.php:2]: :no_entry_sign: Missing file doc comment

<sub>hash: bd42ae1ae03ead84662a1c995475aae2</sub>

File: src/Tribe/RSVP/Early_Access/Update_Notice.php [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+namespace Tribe\Tickets\RSVP\Early_Access;
+
+use Tribe__Tickets__Main;
+use Tribe__Settings;
+use Tribe__Admin__Notices;
+
+class Update_Notice {
+
```
Comment (tr1b0t) [inline src/Tribe/RSVP/Early_Access/Update_Notice.php:10]: :no_entry_sign: Missing doc comment for class Update_Notice

<sub>hash: be029522be24735fb56fce87ce4be201</sub>

File: src/Tribe/RSVP/Early_Access/Update_Notice.php [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+	/**
+	 * @var Early_Access
+	 */
+	private $early_access;
+
+	/**
+	 * The minimum version of Event Tickets needed to
+	 * display the update notice.
+	 *`
+	 * @since TBD
```
Comment (tr1b0t) [inline src/Tribe/RSVP/Early_Access/Update_Notice.php:20]: :no_entry_sign: Expected 1 space after asterisk; 0 found

<sub>hash: 9b559fc652e8e8ecb485abb21c6c5846</sub>

File: src/Tribe/RSVP/Early_Access/Update_Notice.php [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+	 */
```
Comment (tr1b0t) [inline src/Tribe/RSVP/Early_Access/Update_Notice.php:21]: :no_entry_sign: There must be exactly one blank line before the tags in a doc comment

<sub>hash: e22016b1b417ddf7140e6f4cceb19e12</sub>

File: src/Tribe/RSVP/Early_Access/Update_Notice.php [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+	private $minimum_version_to_show = 'TBD';
```
Comment (tr1b0t) [inline src/Tribe/RSVP/Early_Access/Update_Notice.php:22]: :no_entry_sign: Missing @var tag in member variable comment

<sub>hash: cbffefc2e7433d4520aa33a0b501990b</sub>

File: src/Tribe/RSVP/Early_Access/Update_Notice.php [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+	/**
+	 * Update_Notice constructor.
+	 *
+	 * @since TBD
+	 *
+	 * @param Early_Access $early_access
+	 */
```
Comment (tr1b0t) [inline src/Tribe/RSVP/Early_Access/Update_Notice.php:30]: :no_entry_sign: Missing parameter comment

<sub>hash: 40bd0f12a619b817a25a81cbfab83bbb</sub>

File: src/Tribe/RSVP/Early_Access/Update_Notice.php [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+	 * @return bool Whether the update notice was enqueued or not.
+	 */
+	public function maybe_display_update_notice() {
+		// Early bail: Notice should not show up.
+		if ( ! $this->should_display_update_notice() ) {
+			return false;
+		}
+
+		// Bail if user cannot change settings
+		if ( ! current_user_can( Tribe__Settings::instance()->requiredCap ) ) {
```
Comment (tr1b0t) [inline src/Tribe/RSVP/Early_Access/Update_Notice.php:49]: :no_entry_sign: Inline comments must end in full-stops, exclamation marks, or question marks

<sub>hash: 25c7eae200b1fd28a836c94102485361</sub>

File: src/Tribe/RSVP/Early_Access/Update_Notice.php [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+
+		$has_minimum_version = version_compare( Tribe__Tickets__Main::VERSION, $minimum_version_to_show, '>=' );
+		$has_early_access    = $this->early_access->is_rsvp_early_access();
+
+		$should_display = $has_early_access || $has_minimum_version;
+
+		return (bool) apply_filters( 'tribe_tickets_should_display_new_rsvp_notice', $should_display );
+	}
+}
\ No newline at end of file
```
Comment (tr1b0t) [inline src/Tribe/RSVP/Early_Access/Update_Notice.php:87]: :no_entry_sign: File must end with a newline character

<sub>hash: c3d50a1fdc22474652b3da8b6c496023</sub>

File: src/Tribe/RSVP/Service_Provider.php [hunk 9]
```diff
@@ -48,6 +49,9 @@ private function register_early_access() {
 		add_filter( 'tribe_events_tickets_template_tickets/rsvp.php', [
 			$this->container->make( Template::class ),
 			'override_template'
+		add_action( 'admin_init', [
```
Comment (tr1b0t) [inline src/Tribe/RSVP/Service_Provider.php:52]: * :no_entry_sign: PHP syntax error: syntax error, unexpected 'add_action' (T_STRING), expecting ']'
* :no_entry_sign: Opening parenthesis of a multi-line function call must be the last content on the line
* :no_entry_sign: Only one argument is allowed per line in a multi-line function call

<sub>hash: b840f44640ec9a66fbd53375d29fc224</sub>

**Comment by sc0ttkclark** [comment]
Replaced with #1683

**Pull request closed without merging**